### PR TITLE
Adds dotnet SSO sample using CloudAdapter

### LINKED ISSUE #2811: SSO and Skills sample referred to in docs should be core samples not experimental
The article:

https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-authentication-sso?view=azure-bot-service-4.0&tabs=csharp%2Ceml

refers to this sample:

https://github.com/microsoft/BotBuilder-Samples/tree/main/experimental/sso-with-skills

This sample should be in the core sample set (in the samples folder) rather than in the experimental folder if it is being referenced in this way. Experimental is a intended as a temporary home for experimental work.

And we would ultimately need this sample in all the supported languages.

The code of the sample looks pretty OK so this is just really a matter of moving it. Perhaps we need an update to the readme and a check for other minor consistency issues.

### PR DESCRIPTION
**Pull request by gabog**
Fixes #2811

Created sample 82.skills-sso-cloudadapter that demonstrates how to implement SSO between a root bot and as kill using cloud adapter. 

The [readme](https://github.com/microsoft/BotBuilder-Samples/tree/gabog/SSOSampleWithCloudAdapterDotNet/samples/csharp_dotnetcore/82.skills-sso-cloudadapter#readme) file explains how to get the sample up and running.

The samples is based on 81.skills-skilldialog and has been modified to show how to do SSO with skills and dialogs.

### CODE REVIEW

File: samples/csharp_dotnetcore/82.skills-sso-cloudadapter/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+- Update the `BotFrameworkSkills` section in [RootBot/appsettings.json](RootBot/appsettings.json) with the app ID for the skill you created in the previous step.
+- Create a bot registration in the azure portal for the **RootBot** and update [RootBot/appsettings.json](RootBot/appsettings.json) with the `MicrosoftAppId` and `MicrosoftAppPassword` of the new bot registration.
+- Add the **RootBot** `MicrosoftAppId` to the `AllowedCallers` list in [SkillBot/appsettings.json](SkillBot/appsettings.json).
+- Setup the 2 Azure AD applications for SSO as per steps given in [SkillBot Azure AD](https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-authentication-sso?view=azure-bot-service-4.0&tabs=csharp%2Ceml#create-the-azure-ad-identity-for-skillbot) and [RootBot Azure AD](https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-authentication-sso?view=azure-bot-service-4.0&tabs=csharp%2Ceml#create-the-azure-ad-identity-for-rootbot). You will end up with 2 Azure AD applications - one for the skill consumer and one for the skill.
```
Comment (msomanathan) [inline samples/csharp_dotnetcore/82.skills-sso-cloudadapter/README.md:42]: This step links to a page that asks user to create a new app registration with the additional API permissions. So, should the steps above to create **SkillBot** and **RootBot** refer to this link directly, and this step can be removed?

File: samples/csharp_dotnetcore/82.skills-sso-cloudadapter/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+- Create an Azure Active Directory v2 connection in the bot registration for the **SkillBot** and fill in values from the Azure Active Directory v2 application created for SSO, as per the [docs](https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-authentication-sso?view=azure-bot-service-4.0&tabs=srb%2Ccsharp#create-azure-ad-connection-1). Update [SkillBot/appsettings.json](SkillBot/appsettings.json) with the `ConnectionName`
```
Comment (msomanathan) [inline samples/csharp_dotnetcore/82.skills-sso-cloudadapter/README.md:43]: This step was confusing as to which Azure resource has the OAuth connection setting. Could we explicitly specify that the Azure Bot resource has the connection? Initially, I misunderstood this to be the app registration resource.

**Review by msomanathan — APPROVED** [review]
I tested the functionality and SSO works fine with skills. I just made a couple of comments for the readme documentation.

File: samples/csharp_dotnetcore/82.skills-sso-cloudadapter/README.md (hunk 1 above)
Comment (gabog) [reply samples/csharp_dotnetcore/82.skills-sso-cloudadapter/README.md]: Reworded the readme to make it a bit more clear

File: samples/csharp_dotnetcore/82.skills-sso-cloudadapter/README.md (hunk 2 above)
Comment (gabog) [reply samples/csharp_dotnetcore/82.skills-sso-cloudadapter/README.md]: Reworded the readme to make it a bit more clear

**Pull request merged**